packages/builtin/array.pony

```diff
@@ -63,11 +63,11 @@ class Array[A] is Seq[A]
     """
     _ptr._offset(from_offset)._copy_to(ptr._offset(to_offset), copy_len)
 
-  fun cpointer(): Pointer[A] tag =>
+  fun cpointer(offset: USize = 0): Pointer[A] tag =>
     """
     Return the underlying C-style pointer.
     """
-    _ptr
+    _ptr._offset(offset)
```

Comment:
> Maybe someone can access memory they shouldn't have access to via the combination of `cpointer` and `from_cpointer`?

No, this isn't possible, either currently or after this PR.  The `cpointer` method gives you a `Pointer tag` and `from_cpointer` requires a `Pointer ref`.  There's no way to escalate from a `tag` to a `ref` without passing it through an FFI function, so memory safety is still held (for packages that can't or don't use FFI).

That is, it's currently impossible to do the unsafe scenario you describe, and it will still be impossible after this PR.  So I think it's okay.